src/cpp/ext/csm/metadata_exchange.cc

```diff
@@ -397,7 +316,8 @@ ServiceMeshLabelsInjector::ServiceMeshLabelsInjector(
   // from the peer.
   local_labels_.emplace_back(kCanonicalServiceAttribute,
                              canonical_service_value);
-  local_labels_.emplace_back(kMeshIdAttribute, GetMeshId());
+  local_labels_.emplace_back(
+      kMeshIdAttribute, grpc_core::GetEnv("CSM_MESH_ID").value_or("unknown"));
```

Comment:
Do we have a test somewhere that shows the value of this env var actually propagating through in labels?

Reply:
Yes, we do, there's an existing test that verifies that the `csm_mesh_id` label is populated.